### ISSUE TITLE
Unnecessary quotes

### ISSUE THREAD

**Issue by jkarni** [hvr/cassava#86]
```
Data.Csv> encode [("hi", "hi world")]
"hi,\"hi world\"\r\n"
```

My understanding is that [RFC 4180](http://tools.ietf.org/html/rfc4180) allows for quotes in fields, but they are only _required_ when escaping the delimiter. I don't quite understand why a space (which is not a delimiter in default encode options) causes the field to be quoted, though - that seems like inconsistent and somewhat unpredictable behaviour.

(Relevant code [here](https://github.com/tibbe/cassava/blob/master/Data/Csv/Encoding.hs#L230) - the `== sp` case seems unnecessary.)

Should I send a PR to either make this behaviour optional, or remove it entirely? If so, which would you prefer?

**Comment by jkarni** [comment]
And if the behaviour should be optional, I imagine I should just add another constructor to the quoting added in #80 ?

**Comment by tibbe** [comment]
You're right that the RFC doesn't require it. Embarrassingly enough I don't remember why I added it. It doesn't hurt, in that you're allowed to over-quote. I'd be fine to just remove it though. Perhaps you could use `git blame` a bit on that code and see if you can figure out why it's there? Perhaps as a result of fixing some bug someone filed.

**Comment by luke-clifton** [comment]
@tibbe: the commit that introduces this behaviour is https://github.com/hvr/cassava/commit/8f87940ccdd775bb4af623348fc986baf68eabe1#diff-e0fe9164ce56b2f7e2d8c698089b81c9R458

That was on August 21st, 2012.

The comment mentions fixing a bug in escape. There are however no GitHub issues that predate that commit. So I don't know why that was considered a bug.

Pinging #122 and #118

**Comment by tibbe** [comment]
I'm fine removing it. @hvr?

**Comment by hvr** [comment]
@tibbe sounds good to me (but I'll test this empirically a bit to convince myself since I'm paranoid)

Btw, I've been comparing the behaviour to the Python csv module, and I notice that Python has a 3rd encoding option:

> `csv.QUOTE_NONNUMERIC`
> Instructs writer objects to quote all non-numeric fields.

did you consider adding support for that mode in `cassava` or did you decide deliberately against it?

**Comment by luke-clifton** [comment]
Just completing the loop here. https://github.com/hvr/cassava/pull/122/files adds a new mode which is the new default, that behaves the same as the current default, and changes the minimal mode to be the actual minimal mode which is as it was before the above commit.

**Comment by tibbe** [comment]
The non-numeric thing my be tricky because IIRC we do escaping at a point where we don't know what we're escaping (e.g. if it's a number or not).

**Comment by hvr** [comment]
resolved by 2fd71dd
